### PR TITLE
Fix message routing service not delete

### PR DESCRIPTION
**Pull request by gkrenn**
Fix 2 bugs where msg routing service not deleted: 
- After deletion of dynakube object, msgrouting not destroyed
- After disabling of routing in CR-file, msgrouting not destroyed

### CODE REVIEW

File: controllers/capability/routing/reconciler.go [hunk 1]
```diff
@@ -94,11 +95,16 @@ func (r *Reconciler) Reconcile() (update bool, err error) {
 }
 
 func (r *Reconciler) createServiceIfNotExists() (bool, error) {
-	service := createService(r.Instance, module)
-	err := r.Get(context.TODO(), client.ObjectKey{Name: service.Name, Namespace: service.Namespace}, &service)
+	service := createService(r.Instance, Module)
+
+	if err := controllerutil.SetControllerReference(r.Instance, service, r.Scheme()); err != nil {
```
Comment (lrgar) [inline controllers/capability/routing/reconciler.go:100]: Technically you only need to set this reference when creating, i.e., you didn't find it.

As it's now, you're doing this operation even if you create it or not.

tldr: put it inside the if below.

File: controllers/dynakube/dynakube_controller.go [hunk 2]
```diff
@@ -260,6 +260,16 @@ func (r *ReconcileDynaKube) reconcileRouting(rec *utils.Reconciliation, dtc dtcl
 		if err := r.ensureDeleted(&sts); rec.Error(err) {
 			return false
 		}
+
+		ser := corev1.Service{
```
Comment (lrgar) [inline controllers/dynakube/dynakube_controller.go:264]: I don't actually care, but just so you know that the short name for services in Kubernetes is `svc`.

**Review by lrgar — CHANGES_REQUESTED** [review]

**Review by lrgar — APPROVED** [review]
